### PR TITLE
Add full clean checks for event, job and volunteer

### LINKED ISSUE #797: Implement full clean for all fields in unit tests
## Description
All fields should be checked in unit tests via full clean method to know if they actually raise errors.

## Mocks
N/A

### Update [Required]
- [ ] implement full clean check for all fields in unit tests

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
6-7 hours

### PR DESCRIPTION
**Pull request by Monal5031**
# Description
This PR includes full clean checks on all fields of event, job and volunteer but we still need to wait for extra checks PR to avoid editing same thing in different PR (utils.py of shift app).
full clean of shift is done in #726 and for admin in #808 

Fixes #797 

# Type of Change:
**Delete irrelevant options.**

- Code
- Quality Assurance
- User Interface

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
```
python manage.py test event.tests.test_model -v 2
python manage.py test job.tests.test_model -v 2
python manage.py test volunteer.tests.test_unit -v 2
```

# Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by Monal5031** [comment]
@MehaKaushik @naman1901 @ayushgarg1804 @pavankm Done with changes please review.

File: vms/event/tests/test_model.py [hunk 1]
```diff
@@ -38,14 +38,31 @@ def test_valid_model_create(self):
...
+    #     self.assertRaisesRegexp(ValidationError, EventsPage.ENTER_VALID_VALUE, event.full_clean)
+
+    # def test_invalid_start_date_in_model_create(self):
+    #     """
+    #      Database test for model creation with invalid end date.
+    #     """
+    #     This test need to be uncommented after clean method is defined for model.
+    #     event_data = ['event-name', '2050-05-21', '2013-05-24']
+    #     event = create_event_with_details(event_data)
+    #     self.assertRaisesRegexp(ValidationError, EventsPage.ENTER_VALID_VALUE, event.full_clean)
```
[the remark this answers is not in the document]
Comment (Monal5031) [reply vms/event/tests/test_model.py]: Clean methods for model needed for the date checks to be uncommented. I've added a comment too in code and updated #743 for uncommenting and fixing these.

**Comment by Monal5031** [comment]
ping @MehaKaushik @naman1901 Done with changes. please review.

File: vms/event/tests/test_model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ayushgarg1804) [reply vms/event/tests/test_model.py]: oh ok

**Pull request merged**